### PR TITLE
cb-230 integration for `<Approvals />`

### PR DESCRIPTION
**Pull request by Andresbl123**
Integrate the service related to “approvals”

branch: feature-CDB-44

Endpoint: http://localhost:8077/icorebanking-vi-crediboard-query-process-service/api//credit-request/aprovals/{creditRequestId}

AC:SearchAllAprovalsById

x-business-unit: test

https://gitlab.inube.com.co/icorebanking-vi/icorebanking-vi-crediboard-query-process-service/-/tree/feature-CDB-44?ref_type=heads

Documentation:

https://gitlab.inube.com.co/icorebanking-vi/icorebanking-vi-crediboard-query-process-service/-/blob/feature-CDB-44/api/rest/swagger.json?ref_type=heads

image

image

### CODE REVIEW

File: src/pages/board/outlets/financialReporting/Approvals/config.tsx [hunk 1]
```diff
@@ -7,12 +7,20 @@ import {
...
 import { IEntries } from "@components/data/TableBoard/types";
+import { IApprovals } from "./types";
 
 const handleData = (data: IEntries) => {
   console.log(data, "function that receives data");
 };
 
+export const apiUrl = {
+  approvals:
+    "http://192.168.31.62:8077/icorebanking-vi-crediboard-query-process-service/api/credit-requests/aprovals/",
```
Comment (diegoAndres01) [inline src/pages/board/outlets/financialReporting/Approvals/config.tsx:21]: Create it in an environment variable

File: src/utils/hooks/useFetch.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from "react";
+
+export function useFetch<T>(apiUrl: string, retryFlag: boolean) {
```
Comment (diegoAndres01) [inline src/utils/hooks/useFetch.ts:3]: Create it in the service folder

**Review by diegoAndres01 — CHANGES_REQUESTED** [review]

**Review by diegoAndres01 — APPROVED** [review]

File: src/config/pages/board/oulet/financialReporting/configApprovals.tsx [hunk 3]
```diff
@@ -7,11 +7,13 @@ import {
   MdWarningAmber,
```
Comment (carin91) [inline src/config/pages/board/oulet/financialReporting/configApprovals.tsx:7]: Rename the folder "oulet" to "outlet"

File: src/config/pages/board/oulet/financialReporting/configApprovals.tsx [hunk 4]
```diff
@@ -84,6 +84,14 @@ interface TagElement {
   props: TagProps;
 }
 
+export const infoItems = [
+  { icon: <MdCheck />, text: "Aprobado" },
```
Comment (carin91) [inline src/config/pages/board/oulet/financialReporting/configApprovals.tsx:88]: It is necessary to assign a color and an icon to each of the five possible values for this field. Sergio will add the additional values to the mockups in Figma

**Review by carin91 — APPROVED** [review]

**Review by carin91 — APPROVED** [review]

**Pull request merged**